augur/api/view/routes.py

```diff
@@ -223,124 +286,99 @@ def user_settings():
+    
+    # Cannot find the repo
+    return render_message("Repository not found", f"Augur was unable to locate repository with ID: {id}")
 
 """ ----------------------------------------------------------------
-default:
-table:
-    This route returns the groups view for the logged in user.
+group page:
+    This route returns a list of repositories associated with the
+    currently logged in user.
 """
 @app.route('/user/groups/')
 @login_required
 def user_groups_view():
-    params = {}
-
-    pagination_offset = get_value("frontend", "pagination_offset")
-
-    params = {}
-    
-    if query := request.args.get('q'):
-        params["search"] = query
-
-    if sort := request.args.get('s'):
-        params["sort"] = sort
-
-    rev = request.args.get('r')
-    if rev is not None:
-        if rev == "False":
-            rev = False
-            params["direction"] = "ASC"
-        elif rev == "True":
-            rev = True
-            params["direction"] = "DESC"
-
     try:
-        activepage = int(request.args.get('p')) if 'p' in request.args else 0
-    except:
-        activepage = 0
-
-    (groups, status) = current_user.get_groups_info(**params)
-
-    # if not groups and not query:
-    #     return render_message("No Groups Defined", "You do not have any groups defined, you can add groups on you profile page.")
-    # elif not groups:
-    #     return render_message("No Matching Groups", "Your search did not match any group names.")
-
-    page_count = len(groups)
-    page_count //= pagination_offset
-    current_page_start = activepage * pagination_offset
-    current_page_end = current_page_start + pagination_offset
-
-    groups = groups[current_page_start : current_page_end]
-
-    return render_module("groups-table", title="Groups", groups=groups, query_key=query, activePage=activepage, pages=page_count, offset=pagination_offset, PS="user_groups_view", reverse = rev, sorting = sort)
-
+        groups, favorites = current_user.get_groups()
+        
+        for g in favorites:
+            if g in groups:
+                groups.remove(g)
+        
+        # Get the repo counts for all the groups and favorites
+        repoGroups = []
+        for g in groups:
+            repoGroups.append({
+                "name": g.name,
+                "repos": current_user.get_group_repos(db_session, g.name),
+                "favorited": False
+            })
+        
+        favGroups = []
+        for g in favorites:
+            favGroups.append({
+                "name": g.name,
+                "repos": current_user.get_group_repos(db_session, g.name),
+                "favorited": True
+            })
+        
+        # Sort the groups by name
+        repoGroups.sort(key=lambda x: x["name"])
+        favGroups.sort(key=lambda x: x["name"])
+        
+        # Return the page
+        return render_module("groups-table", groups=repoGroups, favorites=favGroups)
```

Comment:
The `groups-table` template does not take a `favorites` argument, and that file was not updated in this PR to add it.